content/docs/learning-guide/tutorials/custom-tools/shape-example-cpp.md

```diff
@@ -0,0 +1,383 @@
+   
+2. Later, you will query types of shape components, add buttons to this widget, and add signal listeners to the buttons.
+
+3. Finally, set the layout of `shapeButtons` to `gridLayout`, and add `shapeButtons` to `mainLayout`. 
+
+```cpp
+// 1
+QWidget* shapeButtons = new QWidget(this);
+QGridLayout* gridLayout = new QGridLayout();
+
+// 2
+
+// 3
+shapeButtons->setLayout(gridLayout);
+mainLayout->addWidget(shapeButtons);
+```
+
+### Query types of Shape components
+
+First, query the types of shape components by specifying a list of provided services. In O3DE, all components have a list of provided services, that you can query from. In this example, Shape components provide the "ShapeService". 
+
+1. Create a `AzToolsFramework::EntityCompositionRequests::ComponentServicesList` called `providedServices` and add "ShapeService". 
+
+2. Get a list of component types. Call `AzToolsFramework::EditorComponentAPIBus::BroadcastResult(...)` and dispatch `AzToolsFramework::EditorComponentAPIRequests::FindComponentTypeIdsByService`, which takes a list of services to include (`providedServices`) and services to exclude (an empty list) and stores the component types to `typeIds`.
+
+```cpp
+AzToolsFramework::EntityCompositionRequests::ComponentServicesList providedServices;
+providedServices.push_back(AZ_CRC_CE("ShapeService"));
+
+AZ::ComponentTypeList typeIds;
+AzToolsFramework::EditorComponentAPIBus::BroadcastResult(
+    typeIds,
+    &AzToolsFramework::EditorComponentAPIRequests::FindComponentTypeIdsByService,
+    providedServices,
+    AzToolsFramework::EntityCompositionRequests::ComponentServicesList()
+);
+```
+
+
+Next, get the names of the components to add to the buttons. 
+
+1. Get a list of component names. Call `AzToolsFramework::EditorComponentAPIBus::BroadcastResult(...)` and dispatch `AzToolsFramework::EditorComponentAPIRequests::FindComponentTypeNames`, which takes the list of component types (`typeIds`) and stores the associated of component names to `componentNames`.
+
+```cpp
+AZStd::vector<AZStd::string> componentNames;
+AzToolsFramework::EditorComponentAPIBus::BroadcastResult(
+    componentNames,
+    &AzToolsFramework::EditorComponentAPIRequests::FindComponentTypeNames,
+    typeIds
+);
+```
+
+{{ note }}
+You can query a list of services that components provide by calling the component's `GetProvidedServices()` method.
+{{ /note }}
+
+### Add buttons
+
+Next, you will loop through the list of components, create buttons, and add them to the `gridLayout`. Refer to the following snippet in `ShapeEXampleQWidget.cpp`.
+
+For each component:
+
+1. Store the component's name and type from the list of component names and types that you queried earlier.
+
+2. Create a button using the QButton object called `shapeButton`. 
+   - Pass in `name` to add the component's name onto the button. 
+   - Set the button minimum height to `40` by using `setMinimumHeight(...)`. 
```

Comment:
Can remove this line since we aren't using the icons anymore we don't need to set the minimum height.